packages/forms/resources/views/components/checkbox-list.blade.php

```diff
@@ -58,15 +58,23 @@
             },
 
             toggleAllCheckboxes: function () {
+                updatedStates = []
                 state = ! this.areAllCheckboxesChecked
 
                 this.visibleCheckboxListOptions.forEach((checkboxLabel) => {
                     checkbox = checkboxLabel.querySelector('input[type=checkbox]')
 
                     checkbox.checked = state
-                    checkbox.dispatchEvent(new Event('change'))
+
+                    if (state) {
+                        updatedStates.push(checkbox.value);
+                    }
+
+                    this.checkIfAllCheckboxesAreChecked()
                 })
 
+                @this.set(@js($getStatePath()), updatedStates);
```

Comment:
Let's use `$wire` here please for consistency

Reply:
done